# Notebook da primeira aula de programação

# New Section

## Configuração do setup(import de bibliotecas e classes)

In [0]:
from requests import get
from bs4 import BeautifulSoup
import pandas as pd

## Requesição do site e edição do html

In [0]:
# funcao que recebe um int e retorna a pagina i do spotted
spotted_link = lambda i: "https://spottedsunicamp.herokuapp.com/spotted/list_spotteds/?page=%d"%i

In [0]:
# requisição do site
get_site = get(spotted_link(1))
get_site.text

'<!doctype html>\n<html>\n\n<head>\n\n\t<!-- Load project\'s staticfiles -->\n\t\n\t\n\t<!-- facebook meta tags -->\n\t<meta property="fb:app_id"          content="1204077889712577" />\n\t<meta property="og:type"            content="website" />\n\t<meta property="og:url"             content="https://spottedsunicamp.herokuapp.com/" />\n\t<meta property="og:title"           content="Spotted Unicamp" />\n\t<meta property="og:image"           content="https://spottedsunicamp.herokuapp.com/static/main/img/logo.2f24221441b7.png" />\n\t<meta property="og:description"    content="Because true love dies. Faça seu primeiro login para se registrar em nosso sistema e ser marcado em spotteds." />\n\n\t<meta charset="utf-8">\n\t<meta http-equiv="X-UA-Compatible" content="IE=edge">\n\t<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0">\n\t<meta name="description" content="Because true love never dies">\n\t<link rel="icon" href="/static/main/img/favicon.7f26403ba2

In [0]:
# celula para achar os spotteds
soup = BeautifulSoup(get_site.text)
l = list(map(str, soup.findAll("div", {"class":"description"})))
l

['<div class="description">\n<p>\n                            Juliana Lopes você é muito gata e gente fina. Quando fiquei feliz em descobrir que vc tava na minha sala as aulas foram suspensas. Como proceder? O corona acabou com qualquer chance que eu tinha\n                        </p>\n</div>',
 '<div class="description">\n<p>\n                            Como que faço pra resistir à tentação de chamar aquela pessoa que gosto mas não tá nem aí pra mim? Dói tanto ser rejeitado por alguém que tive tão boas conversas um dia... A dor de "quase" ter rolado alguma coisa é foda... E ela parece um anjo...\n                        </p>\n</div>',
 '<div class="description">\n<p>\n                            É certo jogar tudo fora por causa de um erro pequeno?\n                        </p>\n</div>',
 '<div class="description">\n<p>\n                            Venho por meio desse spotted pra enaltecer meu amigo Pedro Rissardi que é um amigo FANTÁSTICO s2 E que independente da hora ou do que eu

In [0]:
# edico dos spotteds de uma pagina para só termos os textos sem a formatacao de html

spotteds = []
for i in l: 
  spotteds.append(i.replace("""<div class="description">\n<p>""", "").replace("</p>", "").replace("</div>", "").replace("\n                        \n", "").replace("\n                            ", ""))
spotteds

## Atribuição das páginas para cada pessoa

In [0]:
(5910/11)*5
for i in range(1, 12):
  print((5910/11)*i)

537.2727272727273
1074.5454545454545
1611.8181818181818
2149.090909090909
2686.363636363636
3223.6363636363635
3760.909090909091
4298.181818181818
4835.454545454545
5372.727272727272
5910.0


In [0]:
# adicionei essa celula(não tem na aula gravada)
# Ela serve para automatizar o print das páginas para cada pessoa, na aula fiz na mão mas daria para fazer dessa forma tbm
# Se tiverem dúvida nessa célula posso abordar da próxima aula ou podem perguntar no grupo

nomes = ["Baralho", "Andreis", "Paulo", "Fernanda", "Gabriel", "Vinicius", "Antonio", "Lucca", "Pedro", "Mariana", "Mateus"]
inicial = 1
for i in range (1, 12):
  val = (5910/11)*i
  print("{} = {} - {}".format(nomes[i-1], int(inicial), int(val)))
  inicial = val + 1

Baralho = 1 - 537
Andreis = 538 - 1074
Paulo = 1075 - 1611
Fernanda = 1612 - 2149
Gabriel = 2150 - 2686
Vinicius = 2687 - 3223
Antonio = 3224 - 3760
Lucca = 3761 - 4298
Pedro = 4299 - 4835
Mariana = 4836 - 5372
Mateus = 5373 - 5910


Baralho = 1 - 537

Andreis = 538 - 1074

Paulo = 1075 - 1611

Fernanda = 1612 - 2149

Gabriel = 2150 - 2686

Vinicius = 2687 - 3223

Antonio = 3224 - 3760

Lucca = 3761 - 4298

Pedro = 4299 - 4835

Mariana = 4836 - 5372

Mateus = 5373 - 5910


## Formatação dos dados para csv

In [0]:
# criacao de um dataframe e indexacao de informacoes 

df = pd.DataFrame(columns=["Nomes"])

nomes = ["Baralho", "Andreis", "Paulo", "Fernanda", "Gabriel", "Vinicius", "Antonio", "Lucca", "Pedro", "Mariana", "Mateus"]

for i in nomes:
  df = df.append({"Nomes":i}, ignore_index=True)

df

Nomes
0    Baralho
1    Andreis
2      Paulo
3   Fernanda
4    Gabriel
5   Vinicius
6    Antonio
7      Lucca
8      Pedro
9    Mariana
10    Mateus

In [0]:
#criando um arquivo para guardarmos as informacoes

f = open("Nomes.csv", "a+")
f.write(df.to_csv(index=False))
f.close()

In [0]:
# lendo as informacoes guardadas

df = pd.read_csv("Nomes.csv")
df

Nomes
0    Baralho
1    Andreis
2      Paulo
3   Fernanda
4    Gabriel
5   Vinicius
6    Antonio
7      Lucca
8      Pedro
9    Mariana
10    Mateus

In [0]:
# exemplo de como vai acabar sendo a requisicao dos spotteds em varias paginas

names = pd.DataFrame(columns=["Names"])
nomes = [["Baralho", "Andreis", "Paulo"], ["Fernanda", "Gabriel", "Vinicius"], ["Antonio", "Lucca", "Pedro", "Mariana", "Mateus"]]
for i in nomes:
  for l in i:
    names = names.append({"Names":l},ignore_index=True)

names

Names
0    Baralho
1    Andreis
2      Paulo
3   Fernanda
4    Gabriel
5   Vinicius
6    Antonio
7      Lucca
8      Pedro
9    Mariana
10    Mateus